api/opencl_runtime_layer.asciidoc

```diff
@@ -16589,6 +16589,39 @@ device occupancy high by avoiding blocking in host code.
 
 [open,refpage='clUpdateMutableCommandsKHR',desc='Modify configuration of mutable-command handles to update behavior for future enqueues',type='protos']
 --
+
+==== Update to Command-Buffer Lifecycle
+
+If this extention is supported, the command-buffer lifecycle is extended to
+include a new state named **Finalized**.
+
+// Image generated from the following mermaid diagram description using
+// https://mermaid.live:
+// ---
+// config:
+//  theme: default
+//  look: classic
+// ---
+// stateDiagram-v2
+//  state if_state <<choice>>
+//  if_state: All ND-range kernels commands' arguments set?
+//  [*] --> Recording: Create
+//  Recording --> if_state: Finalize
+//  if_state --> Finalized: No
+//  if_state --> Executable : Yes
```

Comment:
This "if_state" sounds more like a transition than a state itself. The transition can be from "Finalized" to "Executable" if a command-buffer has all it's arguments set.